styx-api-service/src/main/java/com/spotify/styx/api/util/WorkflowFiltering.java

```diff
@@ -0,0 +1,98 @@
+/*-
+ * -\-\-
+ * Spotify Styx API Service
+ * --
+ * Copyright (C) 2016 - 2022 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.styx.api.util;
+
+import static com.spotify.styx.api.util.QueryParams.DEPLOYMENT_TIME_AFTER;
+import static com.spotify.styx.api.util.QueryParams.DEPLOYMENT_TIME_BEFORE;
+import static com.spotify.styx.api.util.QueryParams.DEPLOYMENT_TYPE;
+import static java.util.stream.Collectors.toList;
+
+import com.spotify.styx.model.Workflow;
+import com.spotify.styx.model.WorkflowConfiguration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Predicate;
+import joptsimple.internal.Strings;
+
+public final class WorkflowFiltering {
+
+  private WorkflowFiltering(){}
+
+  public static List<Workflow> filterWorkflows(
+      Collection<Workflow> workflows, Map<QueryParams, String> paramFilters){
+    List<Predicate> workflowFilters = createWorkflowFilters(paramFilters);
```

Comment:
```suggestion
    if (paramFilters.isEmpty()) {
      // Nothing to filter on
      return workflows;
    }
    List<Predicate> workflowFilters = createWorkflowFilters(paramFilters);
```
Do less work when nothing to filter on